config/routes.rb

```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
   namespace :oauth do
     get "authorize", to: "oauth#authorize"
+    post "consent", to: "oauth#consent"
```

Comment:
Your `callback` should be a `GET` since you aren't pushing information to the app so much as you're requesting an authentication code that it already has. It gets wonky since you're including user information but normally it wouldn't have those and the client identifier & state token are both okay to include in URL params.

Reply:
same with this one, since this isn't a callback that is just retrieving data, its actually consuming the users decision and returning a new code if the user gives permission, I chose to make this a POST. Let me know if with this clarification you still believe this should be a GET.